File: RedSheet.DbEntities/Models/ValueObjective.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ValueObjectives", Schema = "dbo")]
	[Serializable]
	public partial class ValueObjective  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ValueObjectiveId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string ValueObjectiveName { get; set; } 
	[InverseProperty("ValueObjective")]
	public ICollection<ProjectBackground> ProjectBackgrounds { get; set; } 

	[InverseProperty("ValueObjective")]
	public ICollection<NanoScopeToNegotiateObjective> NanoScopeToNegotiateObjectives { get; set; } 


	public ValueObjective()
	{
	this.ProjectBackgrounds = new HashSet<ProjectBackground>();

	this.NanoScopeToNegotiateObjectives = new HashSet<NanoScopeToNegotiateObjective>();

    }

    }
}

File: redsheet-portal/src/app/database-models/value-objective.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectBackground, NanoScopeToNegotiateObjective,  } from './'
export class ValueObjective {
    constructor(valueObjective?: ValueObjective )  {
        let properties = [ "className", "valueObjectiveId", "valueObjectiveName", "projectBackgrounds", "nanoScopeToNegotiateObjectives",];
        for (let property of properties)
            if (valueObjective && valueObjective[property])
                this[property] = valueObjective[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	valueObjectiveId : number =   0 ;
 
    @required()
    @maxLength(200)
	valueObjectiveName : string =   undefined;
	@nested(ProjectBackground)
	projectBackgrounds: ProjectBackground[];

	@nested(NanoScopeToNegotiateObjective)
	nanoScopeToNegotiateObjectives: NanoScopeToNegotiateObjective[];



}
